### ISSUE TITLE
[2020] Revisiting ResNets: Improved Training and Scaling Strategies

### ISSUE THREAD

**Issue by cfiken** [cfiken/paper-reading#125]
## 0. 論文

| タイトル | Revisiting ResNets: Improved Training and Scaling Strategies |
|:-------|:--|
| リンク  | https://arxiv.org/abs/2103.07579 |
| 著者    | Irwan Bello, William Fedus, Xianzhi Du, Ekin D. Cubuk, Aravind Srinivas, Tsung-Yi Lin, Jonathon Shlens, Barret Zoph (Google Brain) |
| 投稿日付 | 2021/3/13 on arXiv |

## 1. どんなもの？

画像モデルでの発展は、モデルのアーキテクチャ、学習(正則化)手法、スケーリング戦略の3つによるものがメインだが、アーキテクチャ以外の2つは修正時に同様の設定でなかったり、古いベースラインと比較する際に新しい手法が加えられていないなど、ちゃんとした比較をされていないという課題がある。
本研究では、学習手法とスケーリング戦略の2つに着目し、幅広く実験を行い有効な手法・戦略を探索。改善した学習方法で従来の ResNet のモデルを変えずに ImageNet top-1 スコアを3%更新した。
これらの改善と、(2018年以降は一般的な)少しのアーキテクチャ修正を取り入れた ResNet-RS を提案。EfficietNets と speed/acc 曲線で2.1~3.3倍(GPU)の改善を達成。今後の研究におけるシンプルなベースラインモデルとして使ってほしいとのこと。

![スクリーンショット 2021-03-23 23 11 40](https://user-images.githubusercontent.com/1255485/112171242-ffa41c80-8c36-11eb-8bce-1e80e41e0d29.png)

スケーリング戦略についても既存研究で行われていた小規模実験からの外挿による戦略での欠陥を実験により指摘、学習の規模によってモデルのスケーリングの方向性は変わる(10 epoch 程度なら width を増やしたほうがよく、100 epoch 超えるなら depth を増やしたほうが良い、等)としている。

## 2. 先行研究と比べてどこがすごい？

- 様々な学習・正則化手法を広く実験し、アーキテクチャを変えずに ResNet の性能を大きく改善。
- スケーリングについて、実験規模によって最適なスケールが変わることを実験で見つけ、小規模実験からの外挿が難しいことを指摘。
- ResNet-RS が EfficientNets を speed/acc 曲線において大きく改善。

## 3. 技術や手法のキモはどこ？

- 学習手法・正則化手法
    - LR Scheduler, training epochs, EMA, label smoothing, stochastic depth, augmentation, dropout, weight decay あたりを調整している
    - ![スクリーンショット 2021-03-24 0 15 50](https://user-images.githubusercontent.com/1255485/112171710-6c1f1b80-8c37-11eb-93e1-3eecd6608c26.png)
        - 学習 epoch 数は正則化導入前に増やしているので精度低下しており、また dropout は weight decay の値を下げることで改善することも確認している
- スケーリング戦略
    - いくつかの実験により多くの知見を得ている
        - FLOPs は大きいモデルではそこまで参考にならず、モデルの width, depth, resolution によって結果は変わる。
        - 最適なモデルサイズは学習規模によって変わる。そのため小規模実験の結果から外挿する方法は適切でない。
        - 少ない epochs では width を、多い epochs では depth を増やすのが良い
    - ![スクリーンショット 2021-03-24 0 41 29](https://user-images.githubusercontent.com/1255485/112174378-b4d7d400-8c39-11eb-99e1-13148fc77fa6.png)
- モデルの改善は次の2つのみ (どちらも現在では一般的なもの)
    - ResNet-D
    - Squeeze-and-Excitation
    - 上記の全ての改善を取り入れたモデルを ResNet-RS としている。

## 4. どうやって有効だと検証した？

まず EfficientNets と比較し、speed/acc 曲線で改善していることを確認。

![スクリーンショット 2021-03-24 0 37 34](https://user-images.githubusercontent.com/1255485/112174253-9c67b980-8c39-11eb-9d55-1ec267ebc9fd.png)

FLOPs やパラメータ数では EfficientNet の方が少ないにも関わらず、学習・推論スピードでは ResNet-RS が上回っている。なお、入力の resolution の関係でメモリ使用量も ResNet-RS の方が少ない(下表)。

![スクリーンショット 2021-03-24 0 43 09](https://user-images.githubusercontent.com/1255485/112174721-fc5e6000-8c39-11eb-8f0d-c69b0ff85cca.png)

他にも次のような実験で結果を残している:

- 大量データでの実験として、NoisyStudent と同じ設定で ImageNet + 130M の pseudo labeling データで学習
    - EfficientNet-B5 と同等のスコアを5.5倍(V100)のスピードで達成
- Self-supervised な手法 (SimCLR) とも下位タスクの精度で比較、同等かやや良いくらいの精度を記録
- 動画分類タスク(Kinetic-400) でも実験し、学習手法の改善だけで4%精度を改善した

## 5. 議論はある？

Discussion より

- 異なるアーキテクチャをどう比べたら良いか?
    - 学習手法と分けて比較する
    - FLOPs やパラメータ数で比較している研究が多いが、メモリ消費量やレイテンシで測るべき
- 改善した学習手法はいろんなタスクに使える?
    - ドメインやデータの量による。augmentation とか特に
- スケーリング戦略はいろんなタスクに使える?
    - 学習の規模と、過学習が起こってるかどうかによる。詳しくは Section 6.
    - detection や segmentation タスクなど、resolution がでかい場合は今回実験できてないので分からない。
- アーキテクチャの改善は有効か?
    - Yes だけど学習手法やスケーリング戦略のほうが重要。
